src/plugin/systems.rs

```diff
@@ -525,10 +521,23 @@ pub fn writeback_rigid_bodies(
                                     .inverse()
                                     .to_scale_rotation_translation();
                             let new_rotation = inverse_parent_rotation * interpolated_pos.rotation;
+
+                            #[cfg(feature = "dim3")]
                             let new_translation = inverse_parent_rotation
                                 * interpolated_pos.translation
                                 + inverse_parent_translation;
 
+                            #[cfg(feature = "dim2")]
+                            let mut new_translation = inverse_parent_rotation
+                                * interpolated_pos.translation
+                                + inverse_parent_translation;
```

Comment:
Let’s avoid duplicating the code by silencing the warning instead:
```suggestion

                            #[allow(unused_mut)] // mut is needed in 2D but not in 3D.
                            let mut new_translation = inverse_parent_rotation
                                * interpolated_pos.translation
                                + inverse_parent_translation;
```

Reply:
Gladly, thanks!